### ISSUE TITLE
Old build issue back [docker-compose crash]

### ISSUE THREAD

**Issue by ZEROF** [Shippable/support#3730]
Hello,

We have this issue once again after agent upgrade.

`WARNING: Dependency conflict: an older version of the 'docker-py' package may be polluting the namespace. If you're experiencing crashes, run the following command to remedy the issue: pip uninstall docker-py; pip uninstall docker; pip install docker Creating network "application_default" with the default driver Pulling redis (redis:3.2.9-alpine)... Traceback (most recent call last): File "/usr/local/bin/docker-compose", line 11, in <module>`

I pulled your  Drydock Family : u16 and u14py as well on my local machine and i have same errors, this need fixing ASAP, docker-compose installed to this containers just not working. Tested installing few version of docker-compose too.

I'm out of ideas to be honest. I send mail to @devashish but i didn't get any answer.

Thanks

**Comment by ric03uec** [comment]
@ZEROF we're looking into it

**Comment by kanthchandra** [comment]
Hey @ZEROF I think this seems to be a problem with docker compose version 1.10 onwards as reported [here] (https://github.com/docker/compose/issues/4344#issuecomment-273932670) . So can you try and see if adding this command `pip install docker-compose==1.9.0` works?

**Comment by ZEROF** [comment]
I pushed my own image as solution [ubuntu 16.04, up to date, with gcloud sdk, python, docker-compose, docker etc]. This is 2nd time i had issue with your image, I can't count on 'no stress' with node updates, every time i need to look to the sky and cross my fingers, not really stable images on your side, and how i can use that version of docker compose [1.9] with docker compose file format version 3.1, that just will not work. I didn't want to sound rude, but hope that information shared with you, will help you in the future.


| Compose file format | Docker Engine  |
| ------------------------- | -------------------  |
 3.3 | 17.06.0+ 
3.0 – 3.2 | 1.13.0+
2.2 | 1.13.0+
2.1 | 1.12.0+
2.0 | 1.10.0+
1.0 | 1.9.1+

**Comment by kanthchandra** [comment]
Hey @ZEROF In order to debug this further can you provide me the following details?

1. Can you please provide me the link to the build where the problem first started? I want to compare what changed.
2. Can you also please tell me how did you update your custom node? (manually by running the script or did you allow shippable to initalize?)
3. We support only docker 1.11 and 1.13 but on some of the failed builds that I checked it said that the docker version as 17.05, I'm not sure how that happened and I want to first get to the bottom of that. It looks like build 2535 which had docker 1.13 ran on `drydock/u14pytall:v5.5.1` passed while 2536 which had docker 17.05 and ran on `drydock/u14pytall:v5.5.1` failed. One more difference is 2535 ran using agent genexec:v5.7.1 and 2536 ran on agent genexec:v5.7.2. There are two differences here, now we have to figure what caused the actual problem.

**Comment by ZEROF** [comment]
1. 2536
2. For security I update all packages on node [more then 30 of them], then i pushed on 'initialize node' to be sure that node will have right version of docker installed, i learned that last time, agent will install 1.13 version. 
3. I know, because of that we didn't update our compose file version, we use version 'older' 3.1.

Node docker info

| Version | 1.13.0|
| --------- | ----------- |
 API version |  1.25
 Go version |   go1.7.3
 Git commit |   49bf474
 Built | Tue Jan 17 09:58:26 2017
 OS/Arch | linux/amd64


I tested few version of 14py and 16py images just to be sure if we did something on our side. That took few hours going around and testing, i was pissed off and decided to build my image, and all was back to normal. Now i can see old settings work again, you did change something; i guess.

**Comment by kanthchandra** [comment]
For this risk of breaking people who already have their configuration working we dont update any older images of u14pyt or u16pyt. Although I'm not sure how the problem appeared and disappeared I can assure you that we dint change anything on `drydock/u14pytall:v5.5.1` or `drydock/u16pytall:v5.5.1`. These images were last updated 3 months ago.

What we did was we released newer images `drydock/u14pytall:v5.7.3` or `drydock/u16pytall:v5.7.3` and this switch is not automatic, you'll have to go and update the Machine Images in your subscription settings manually to use these images while using shippable nodes.

BTW update the node means you are just merely updating the shippable agent (drydock/genexec image). This agent is just responsible for receiving a build request, booting up your ci container and then posting console logs and cleaning up a few things after your build is complete apart from that this will have no idea as to what is happening inside your build container. So updating shippable agent should have no effect on your build whatsoever.
